src/server/session.ts

```diff
@@ -834,153 +850,178 @@ module ts.server {
 
         exit() {
         }
+        
+        private handlers : Map<ProtocolHandler> = {};

```

Comment:
please split this in a different change. i would like to discuss the design first.